File: src/main/java/br/zom/zup/academy/dominio/repository/AutorRepository.java
package br.zom.zup.academy.dominio.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import br.zom.zup.academy.dominio.modelo.Autor;

@Repository
public interface AutorRepository extends JpaRepository<Autor, Long>{

	Optional<Autor> findByEmail(String email);
}

File: casa-do-codigo-grpc/src/main/kotlin/br/com/zup/dominio/repository/AutorRepository.kt
package br.com.zup.dominio.repository

import br.com.zup.dominio.modelo.Autor
import io.micronaut.data.annotation.Repository
import io.micronaut.data.jpa.repository.JpaRepository

@Repository
interface AutorRepository: JpaRepository<Autor, Long> {

}
